packages/nocodb/src/lib/version-upgrader/ncAttachmentUpgrader_0104002.ts

```diff
@@ -149,6 +156,9 @@ export default async function ({ ncMeta }: NcUpgraderCtx) {
         }
         await Promise.all(updateRecords);
       } catch (e) {
+        // throw the custom timeout error message if applicable
+        throwTimeoutError(e, timeoutErrorInfo);
+
```

Comment:
You can move this to the if condition, for deleted projects we don't have to throw the error

Reply:
The if-condition is inside `throwTimeoutError`.  Since this function could be used in different upgraders, I try to remove the duplicate code as much as possible. If the error code is not `ETIMEDOUT` (e.g. deleted project case), basically it won't throw anything in `throwTimeoutError`.